### PR TITLE
Convert backend/tk.c to D: error handling

### PR DESCRIPTION
**Pull request by JinShil**
I think it will be easier to review this in pieces, so I'm going to try to do this a chunk at a time.

The grand plan is to incrementally convert the code in backend/tk.c to D and move it to backend/mem.d.  When all conversion is finished, we delete tk.c.

### CODE REVIEW

File: src/dmd/backend/tk.c [hunk 1]
```diff
@@ -235,7 +175,7 @@ char *mem_fstrdup(const char *);
 
 #if MEM_NONE
 
-#define mem_inited      1
+//#define mem_inited      1
```
Comment (jacob-carlborg) [inline src/dmd/backend/tk.c]: Why a comment?

**Comment by jacob-carlborg** [comment]
LGTM

File: src/dmd/backend/tk.c (hunk 1 above)
Comment (JinShil) [reply src/dmd/backend/tk.c]: `mem_inited` is defined in `mem.d` now, and declared `extern` in `tk.c`.  I didn't want any symbol conflicts.

Comment (jacob-carlborg) [reply src/dmd/backend/tk.c]: Why not remove it completely?

Comment (JinShil) [reply src/dmd/backend/tk.c]: :blush: Because sometimes I'm an idiot.

**Review by jacob-carlborg — APPROVED** [review]

File: src/dmd/backend/tk.c (hunk 1 above)
Comment (jacob-carlborg) [reply src/dmd/backend/tk.c]: 😊

**Comment by JinShil** [comment]
Closing in favor of https://github.com/dlang/dmd/pull/9592

**Pull request closed without merging**